pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -83,6 +84,7 @@ contract WeightedPool is BaseWeightedPool, InvariantGrowthProtocolFees {
             owner,
             false
         )
+        InvariantGrowthProtocolFees(protocolFeeProvider)
```

Comment:
I had the `ProtocolFeeCache(protocolFeeProvider, ProtocolFeeCache.DELEGATE_PROTOCOL_SWAP_FEES_SENTINEL)` here; with Weighted it will always delegate